DB/src/main/java/io/deephaven/db/tablelogger/ProcessMemoryLogLogger.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2016-2019 Deephaven Data Labs and Patent Pending
+ */
+
+package io.deephaven.db.tablelogger;
+
+import io.deephaven.db.tables.TableDefinition;
+import io.deephaven.db.tables.utils.ColumnsSpecHelper;
+import io.deephaven.db.tables.utils.DBDateTime;
+import io.deephaven.db.tables.utils.DBTimeUtils;
+import io.deephaven.tablelogger.Row;
+import io.deephaven.tablelogger.RowSetter;
+import io.deephaven.tablelogger.TableLoggerImpl2;
+import io.deephaven.tablelogger.WritableRowContainer;
+
+import java.io.IOException;
+
+public class ProcessMemoryLogLogger extends TableLoggerImpl2<ProcessMemoryLogLogger.ISetter> {
+    private static final String TABLE_NAME = "ProcessMemoryLog";
+
+    public ProcessMemoryLogLogger() {
+        super(TABLE_NAME);
+    }
+
+    public static String getDefaultTableName() {
+        return TABLE_NAME;
+    }
+
+    interface ISetter extends WritableRowContainer {
+        void log(
+                final Row.Flags flags,
+                final long intervalStartTime, final long intervalDurationMs,
+                final long totalMemory, final long freeMemory,
+                final long intervalCollection, final long intervalCollectionTimeMs) throws IOException;
+    }
+
+    class DirectSetter extends TableLoggerImpl2.BaseSetter implements ISetter {
+        RowSetter<DBDateTime> IntervalStartTime;
+        RowSetter<Long> IntervalDurationMs;
+        RowSetter<Long> TotalMemory;
+        RowSetter<Long> FreeMemory;
+        RowSetter<Long> IntervalCollections;
+        RowSetter<Long> IntervalCollectionTimeMs;
+
+        DirectSetter() {
+            IntervalStartTime = row.getSetter("IntervalStartTime", DBDateTime.class);
+            IntervalDurationMs = row.getSetter("IntervalDurationMs", long.class);
```

Comment:
I care about the usability of the new table.   My whole point trying to do this is to make something easy to use for median user to be able to be self-served answering the question "Am I having a GC issue".   That column being in nanos gets in the way of that.  So I want to keep it in milliseconds.

If you /really/ care about this this much (I hope not, because I don't see something with specific units as such consistency issue, but you might), we could turn this into a percentage.

Reply:
Btw what we obviously need here is a duration type that has nice rendering (like DBDateTime has nice rendering), and whatever underlying unit.